src/partial-result-stream.js

```diff
@@ -79,6 +80,11 @@ function partialResultStream(requestFn, options) {
         metadata = row.metadata;
       }
 
+      if (pendingRowValues) {
+        row.values = pendingRowValues.concat(row.values);
+        pendingRowValues = null;
+      }
```

Comment:
Unrelated to this change, but below we keep on buffering till chunkedValue is true. That can cause problems since you might buffer a lot. In the pathological case, all but the last PartialResultSet can have this as true. In that case you will buffer the whole stream before yielding anything at all even though you would have already received many complete rows.

Reply:
The `else` clause of the `row.chunkedValue` check is where the rows that are pending are split from the rows that are ready to pass on to the user. Only the columns that were cut short are held back (and become `pendingRowValues`.

Reply:
I think @vkedia  is referring to how the existing code will keep pushing row to rowChunks while row.chunkedValue is true. And in a pathological case that could be every object. Meaning we would collect all values client side, then combine, make rows, and return all rows at once.

What probably has to be done is we should merge chunks as we can and attempt to send rows as we have them. This is different than how this is currently done.

Reply:
@crwilcox yes that is what I meant and the proposed solution seems good.